### ISSUE TITLE
Show compiling log in runtime

### ISSUE THREAD

**Issue by pustovalov** [rails/webpacker#1970]
Can I somehow modify this method https://github.com/rails/webpacker/blame/3-x-stable/lib/webpacker/compiler.rb#L63 to see the real-time log?

Right now I have trouble when webpacker compiling assets on ci and then show this error:

```
Using /home/circleci/main/config/webpacker.yml file for setting up webpack paths
Compiling…
Compilation failed:


Exited with code 1
```

If I run this command with [simple-progress-webpack-plugin](https://github.com/dominique-mueller/simple-progress-webpack-plugin)
```
NODE_ENV=test ./bin/webpack --config config/webpack/test.js
```
I see the real-time log and don't get an error

What I did to prevent this an error:
added `NODE_OPTIONS: --max-old-space-size=3072`

but it did not help

**Comment by david-crespo** [comment]
You can set `webpack_compile_output: true` in the config YAML to include the full webpack output in the Rails log.

**Comment by pustovalov** [comment]
`webpack_compile_output: true` didn't help

```
[~/main]$ bundle exec rake assets:precompile                                                                                                         
yarn install v1.17.3
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.96s.
Compiling…
```

**Comment by david-crespo** [comment]
Weird. Make sure you have it set for the correct environment and you're not overriding it elsewhere.

**Comment by khier996** [comment]
@pustovalov have u resolved your issue? from your last comment it seems that webpacker is stuck at "compile..." step. I have this problem on circleci now. increased the timeout to 35 minutes and it still hangs and times out after 35 minutes

**Comment by pustovalov** [comment]
@khier996 I solved this problem in a different way

`config/webpack/test.js`

```
const SimpleProgressWebpackPlugin = require('simple-progress-webpack-plugin')
environment.plugins.prepend(
  'SimpleProgressWebpackPlugin',
  new SimpleProgressWebpackPlugin({
    format: 'verbose',
  }),
)
```

`config.yml`

```
aliases:
  - &main_container
    environment:
      NODE_ENV: test
      RACK_ENV: test
      RAILS_ENV: test
      WEBPACKER_PRECOMPILE: false

jobs:
  prepare:
    <<: *docker_full
    steps:
      - *restore_assets
      - *restore_assets_cache
      - run:
          name: Precompile Assets
          command: |
            bin/webpack --config "config/webpack/$NODE_ENV.js"
            bundle exec rake assets:precompile
      - save_cache:
          key: *assets_cache_key
          paths:
            - public/assets
            - public/packs-test
            - tmp/cache/assets/sprockets
            - tmp/tkbl_assets

```

**Comment by pustovalov** [comment]
also you try to do this one:
```
NODE_OPTIONS: --max-old-space-size=3400
```

because circleci show you this information:
![](https://docs.google.com/uc?id=1Ov4ocJxPOev3fwgfrBc6udtlrxNBnV6u)

but actually you have a different amount of RAM available

```
Hint: Exit code 137 typically means the process is killed because it was running out of memory
Hint: Check if you can optimize the memory usage in your app
Hint: Max memory usage of this container is 3681169408
```

**Comment by khier996** [comment]
@pustovalov thanks for help! I figured my problem had to do with erb-loader. I found the solution in this[ comment](https://github.com/usabilityhub/rails-erb-loader/issues/47#issuecomment-434138107)

I have a different problem now. Elasticsearch container exits with code 137. This did not happen before we started using webpacker. I know this is not relevant to the original issue, but if you know how to solve, pls let me know, i'll pm you.
